dejavu_bot.py

```diff
@@ -1,111 +1,152 @@
     created_at = channel.created_at
     end = datetime.utcnow().replace(tzinfo=timezone.utc)
     rand_datetime = get_rand_datetime(created_at, end)
 
+    # Fetch a message using the random datetime
     # limit=1 so we only get one message (we could change this later to add more?)
     async for rand_message in channel.history(limit=1, around=rand_datetime):
-        if rand_message.content != '':
-            await create_and_send_response(rand_message, channel, arg)
+        if rand_message.content != "":
+            await create_and_send_response(rand_message, channel, choices.value)
             break
 
+
 def get_rand_datetime(start, end):
     """
     https://stackoverflow.com/questions/553303/generate-a-random-date-between-two-other-dates
 
-    This function will return a random datetime between two datetime 
-    objects.
+    This function will return a random datetime between two datetime objects.
     """
     delta = end - start
     int_delta = (delta.days * 24 * 60 * 60) + delta.seconds
     random_second = randrange(int_delta)
     return start + timedelta(seconds=random_second)
 
-async def create_and_send_response(rand_message, channel, arg):
-    """
-    Creates an image with the message author, content, and creation datetime
-    with a random colored background
-    """
+
+async def create_and_send_response(rand_message, channel, choice):
+
     text = (
-        rand_message.author.name +
-        " said: \n" +
-        rand_message.content +
-        "\nat " +
-        rand_message.created_at.strftime("%Y-%m-%d %I:%M %p")
+        rand_message.author.name
+        + " said: \n"
+        + rand_message.content
+        + "\nat "
+        + rand_message.created_at.strftime("%Y-%m-%d %I:%M %p")
     )
 
-    if arg == 'image':
+    if choice == "text":
+        # Sends the random message with the message author, content,
+        # and creation datetime
+        # `/dejavu text`
+        await channel.send(text)
+    elif choice == "image":
         await create_and_send_image(text, channel)
-    elif arg == 'whosaid':
-        # if the arg is whosaid, store the message details in the bot's who_said_context
-        bot.who_said_context = {
-            'author_id': rand_message.author.id,
-            'content': rand_message.content,
-            'channel': channel
-        }
-        await who_said(rand_message.content, channel)
+    elif choice == "whosaid":
+        # if the choice is whosaid, pass the rand_message and channel to whosaid()
+        await whosaid(rand_message, channel)
     else:
-        # /dejavu text
-        # Just return the random message as text
-        await channel.send(text)
+        await channel.send("Invalid Command.")
+
 
 async def create_and_send_image(text, channel):
     """
-    Handle the case where an image is requested
-    /dejavu image
+    Creates an image with the message author, content, and creation datetime
+    with a random colored background. TODO: Doesn't work right.
```

Comment:
I don't want to merge this until `create_and_send_image` works.

Reply:
:+1: